### ISSUE TITLE
timeline: Create a permalink from a given message

### ISSUE THREAD

**Issue by bnjbvr** [matrix-org/matrix-rust-sdk#3259]

**Comment by bmarty** [comment]
I think @stefanceriu has started working on this on https://github.com/matrix-org/matrix-rust-sdk/pull/3289.

In addition, we will need a way to provide a base URL for the created permalink, i.e. permalinks will not necessarily start by `https://matrix.to/#/`.

The base URL could be provided when requested the permalink, so in the API (Android preference), or in the client Builder (I believe it's iOS preference).

CC @pixlwave

**Comment by pixlwave** [comment]
Yep, if we could just set the base URL once when creating the Client (or technically twice, as the auth service would need it too) that would certainly clean up our code a lot. But obviously if that's a challenge on the Rust side, passing it in the API would also work for us too.

**Comment by stefanceriu** [comment]
I am not, the only thing I did was expose the existing API through FFI
